VulkanHppGenerator.cpp

```diff
@@ -4665,6 +4718,657 @@ std::string VulkanHppGenerator::generateConstexprString( std::string const & str
   return isConstExpression ? ( std::string( "VULKAN_HPP_CONSTEXPR" ) + ( ( containsUnion( structName ) || containsArray( structName ) ) ? "_14 " : " " ) ) : "";
 }
 
+std::pair<std::string, std::string> VulkanHppGenerator::generateConstexprDefinesAndUsings() const
+{
+  auto const constexprFunctionTemplate = std::string{ R"(  template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
+  ${deprecated}VULKAN_HPP_CONSTEXPR auto ${constName}( ${arguments} )
+  {
+    return ${implementation};
+  }
+)" };
+  auto const constexprCallTemplate     = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR auto ${constName} = ${callee}( ${arguments} );
+)" };
+  auto const constexprValueTemplate    = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR ${type} ${constName} = ${value};
+)" };
+  auto const deprecatedAttribute       = std::string{ R"(VULKAN_HPP_DEPRECATED("${reason}") )" };
+
+  auto const constexprUsingTemplate = std::string{ R"(  using VULKAN_HPP_NAMESPACE::${constName};
```

Comment:
Are you referring to the last one, `constexprUsingTemplate`? I can remove it if you insist. 

I personally find using templates nicer than manually concatenating strings because it is clear how the generated code will look at one glance, and since this generator is (supposedly) not run very often, runtime performance is not so critical, even if `replaceWithMap` is run more often than strictly necessary.